### DISCUSSION TITLE
Any solution which uses WindowsAppSDK 1.5.X no longer builds after most recent VisualStudio update

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#18416] [Q&A]
### Current behavior

I get this error on all my solutions which use WindowsAppSDK 1.5.X

`error CS0433: The type 'TextDecorations' exists in both 'Microsoft.WinUI, Version=3.0.0.0, Culture=neutral, PublicKeyToken=de31ebe4ad15742b' and 'Microsoft.Windows.SDK.NET, Version=10.0.19041.38, Culture=neutral, PublicKeyToken=31bf3856ad364e35'`

@jeromelaban appears something broke with most recent VS update ...
Just try to build repo below.

[Minimal.zip](https://github.com/user-attachments/files/17298492/Minimal.zip)

### Expected behavior

My WindowsAppSDK 1.5.X UNO WASM solution should build as before.

**Comment by jeromelaban** [comment]
Thanks for the report. 

I can repro with your sample, but not with standard Uno blank solutions. It's likely a mismatch with WinAppSDK and other packages in your solution. This is also likely a bug in WinAppSDK and MS's other packages.

**Comment by DierkDroth** [comment]
@jeromelaban while trying to work my pay through that absolute show-stopper I created a fresh new UNO solution following [these instructions](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows) (including  uno-check) and now get these warnings when building:

```
1>C:\Program Files\dotnet\sdk\8.0.403\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(204,5): warning NETSDK1198: A publish profile with the name 'win-AnyCPU.pubxml' was not found in the project. Set the PublishProfile property to a valid file name.
1>obj\Debug\net8.0-android\android\AndroidManifest.xml : warning XA4218: Unable to find //manifest/application/uses-library at path: C:\Program Files (x86)\Android\android-sdk\platforms\android-34\optional\androidx.window.extensions.jar
1>obj\Debug\net8.0-android\android\AndroidManifest.xml : warning XA4218: Unable to find //manifest/application/uses-library at path: C:\Program Files (x86)\Android\android-sdk\platforms\android-34\optional\androidx.window.sidecar.jar
```

Is that expected?

**Reply by jeromelaban** [reply]
The first issue is known, we do not have a resolution at this time. The second is fixed in .NET 9.

**Reply by DierkDroth** [reply]
Thanks for your feedback @jeromelaban

**Comment by DierkDroth** [comment]
@jeromelaban the reason why you don't experience the issue with a blank new UNO solution is that the solution is set up in a way where one is forced to use WindowsAppSDK 1.6.X and that can't even be changed (screenshot below). 

-> I have no way to justify that a blank new UNO solution would be working with WindowsAppSDK 1.5.X and have no way to figure out what's going wrong.

Is that what it is, that I now **am forced to update all my UNO solutions to WindowsAppSDK 1.6.X**? 
Meaning I can't even properly maintain/bugfix the builds which I already have deployed?

![image](https://github.com/user-attachments/assets/e6b38536-003d-4b44-a363-20078c8b46ac)

**Reply by jeromelaban** [reply]
This is a question to ask on the WinAppSDK or WinUI repository as this is indeed a breaking change of some sort but you'll notice that if you remove all the uno references from the windows project, you will still have the issue. This is indeed a problem for maintainability.

**Reply by DierkDroth** [reply]
@jeromelaban that actually is not correct! Last night I was about to submit a report on the WinAppSDK repository and created a blank solution with no UNO involved at all, but with WindowsAppSDK 1.5.X and ... surprise, surprise ... there was no issue: no problem whatsoever with 'TextDecorations'!

This made me file my report here on the UNO repo since - for all I know at this point - the original issue above is an UNO issue.

**Reply by jeromelaban** [reply]
Thanks. Make sure to use your sample repro above with the same parameters and remove all the uno-related projects and references, you'll notice the error still exists.

**Reply by DierkDroth** [reply]
You're right @jeromelaban. My apologies.

Dammit, I wished I kept the solution from last night as I was about to post on the WindowsAppSDK repository ... since that was what I did: I removed all UNO dependencies and then it worked! ... go figure ... oh, well ...

**Comment by DierkDroth** [comment]
@jeromelaban to be clear: updating to WindowsAppSDK 1.6.X resolves the issue. However, for obvious reasons this is not a great option to maintain/bugfix for apps which are deployed.

FYI for anyone else struggling with that issue: even upgrading to WindowsAppSDK 1.6.X gave me a headache - since I'm using WebView2 and ran [into this problem](https://learn.microsoft.com/en-us/answers/questions/1818176/error-appx1101-(uwp)-payload-contains-two-or-more). I was able to resolve by explicitly deleting all obj/bin subfolders of my solution(s).

**Comment by DierkDroth** [comment]
@jeromelaban FYI for further reference

https://github.com/microsoft/WindowsAppSDK/issues/4788
